SDK/Hardware/HA4IoT.Hardware.Knx/KnxDigitalJoinEnpoint.cs

```diff
@@ -0,0 +1,51 @@
+﻿using HA4IoT.Contracts.Actuators;
+using HA4IoT.Contracts.Hardware;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace HA4IoT.Hardware.Knx
+{
+    public class KnxDigitalJoinEnpoint : IBinaryStateEndpoint
+    {
+        private readonly string _identifier;
+        private readonly KnxController _knxController;
+
+        public KnxDigitalJoinEnpoint(string identifier, KnxController knxController)
+        {
+            if (identifier == null) throw new ArgumentNullException(nameof(identifier));
+            if (knxController == null) throw new ArgumentNullException(nameof(knxController));
+
+            if (ValidationJoin(identifier))
+            {
+                _identifier = identifier;
+                _knxController = knxController;
+            }
+        }
+
+        public void TurnOn(params IHardwareParameter[] parameters)
+        {
+            _knxController.DigitalJoinOn(_identifier);
+        }
+
+        public void TurnOff(params IHardwareParameter[] parameters)
+        {
+            _knxController.DigitalJoinOff(_identifier);
+        }
+
+        public void Toggle(params IHardwareParameter[] parameters)
```

Comment:
Adding a toggle function is useless because no one will use it. This is an endpoint which means that it has only the absolutely required methods. The actuators which are using that endpoint doing the toggle on their own. They cache the state of the hardware internally. So for your case if you call "TurnOn" at the actuator it will let the Endpoint sending the "1" by using the "TurnOff" method. Of you call "Toggle" at the actuator it knows that the hardware is still on and will send "0" to the hardware using the endpoint. So there is not need for another toggle method on hardware (called endpoints) level. But may I am not understanding what you mean :see_no_evil:.